tutorial/numpy-functions.md

```diff
@@ -0,0 +1,351 @@
+
+    end program prog_test_prod
+    ```
+
+## [mod](https://numpy.org/doc/stable/reference/generated/numpy.mod.html)
+
+-   Supported parameters:
+
+    x1: array_like
+        Dividend array.
+
+    x2: array_like,
+        Divisor array. If x1.shape != x2.shape, they must be
+        broadcastable to a common shape (which becomes the shape of the output).
+
+-   Supported language: fortran.
+
+-   python code:
+
+    ```python
+    from numpy import array, mod
+
+    arr = array([1,2,3,4])
+    res = mod(arr, arr)
+    print("res: ", res)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_test_mod
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr(:)
+    integer(C_INT64_T), allocatable :: res(:)
+
+    allocate(arr(0:4_C_INT64_T - 1_C_INT64_T))
+    arr = [1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T]
+    allocate(res(0:4_C_INT64_T - 1_C_INT64_T))
+    res = modulo(arr,arr)
+    print *, 'res: ' // ' ' , res
+
+    end program prog_test_mod
+    ```
+
+## [matmul](https://numpy.org/doc/stable/reference/generated/numpy.matmul.html)
+
+-   Supported parameters:
+
+    x1, x2: array_like,
+        Input arrays (must be 1d or 2d), scalars not allowed.
+
+-   Supported langauges: fortran (2d arrays only)
+
+-   python code:
+
+    ```python
+    from numpy import array, matmul
+
+    arr = array([[1,2],[3,4]])
+    res = matmul(arr, arr)
+    print("res: ", res)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_test_matmul
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr(:,:)
+    integer(C_INT64_T), allocatable :: res(:,:)
+
+    allocate(arr(0:2_C_INT64_T - 1_C_INT64_T, 0:2_C_INT64_T - 1_C_INT64_T))
+    arr = reshape([[1_C_INT64_T, 2_C_INT64_T], [3_C_INT64_T, 4_C_INT64_T]], &
+          [2_C_INT64_T, 2_C_INT64_T])
+    allocate(res(0:2_C_INT64_T - 1_C_INT64_T, 0:2_C_INT64_T - 1_C_INT64_T))
+    res = matmul(arr,arr)
+    print *, 'res: ' // ' ' , res
+
+    end program prog_test_matmul
+    ```
+
+## [Numpy types](https://numpy.org/devdocs/user/basics.types.html)
+
+-   Supported types : bool, int, int8, int16, int32, int64, float, float32, float64, complex64 and complex128. they can be used as cast functions too.
+
+## Other functions
+
+-   Supported [math functions](https://numpy.org/doc/stable/reference/routines.math.html) (optional parameters are not supported):
+
+    sqrt, abs, sin, cos, exp, log, tan, arcsin, arccos, arctan, arctan2, sinh, cosh, tanh, arcsinh, arccosh and
+    arctanh.
+
+-   Supported [routines array-creation](https://numpy.org/doc/stable/reference/routines.array-creation.html) (fully supported):
+
+    -   empty, full, ones, zeros, arange (`like` parameter is not supported).
+    -   empty_like, full_like, and zeros_like, ones_like (`subok` parameter is not supported).
+    -   rand, randint.
+
+-   others:
+
+    max, min, sum, shape, floor
```

Comment:
Please specify which min/max is used ( [amax](https://numpy.org/doc/stable/reference/generated/numpy.amax.html) )

Reply:
it is a little bit confusing, we support the same argument as amax but we have max as a name for it. same for min function

Reply:
I agree. Thanks for fixing it :+1:

Reply:
Probably the name confusion is due to https://numpy.org/doc/stable/reference/generated/numpy.ndarray.max.html

Reply:
As above I think this used to be fixed. Please mention `amin`/`amax` here instead of `min`/`max`